operator_toolkit/soc_helpers/loader.py

```diff
@@ -0,0 +1,98 @@
+import os
+
+import pandas
+import arrow
+import cabinets
+from influxdb import DataFrameClient
+
+
+CONFIG = cabinets.read("operator_toolkit/soc_helpers/config.yml")
+
+
+class DataLoader:
+
+    def __init__(self, site, start, end, timezone):
+        """Instantiates the DataLoader class.
+
+        :param site: site_id.
+        :param start: start time, in ISO format.
+        :param end: end time, in ISO format.
+        """
+        self.timezone = timezone
+        self.site = site
+        self.start = start
+        self.end = end
+        self.client_config = CONFIG["influxdb_client"]
+        self.data = self.get_final_dataframe(
+            self.extract_data(
+                query=self.compose_query(
+                    query_fields=self.create_whitelist_fields())))
+
+    @staticmethod
+    def create_whitelist_fields():
+        """Check the config file and return the list of fields in the sql query.
+
+        :return: list on mean values in the sql query.
+        :rtype: str.
+        """
+        whitelist_fields = CONFIG["query_fields"]
+        sql_query_fields = []
+        for field in whitelist_fields:
+            sql_query_fields.append(f"mean(/{field}/)")
+        return ", ".join(sql_query_fields)
+
+    def compose_query(self, query_fields):
+        """Composes a sql query for every component.
+
+        :param query_fields: pre-selected fields to be queried to avoid crashing the database.
+        :return: sql query.
+        :rtype: str.
+        """
+        where = f"time > '{self.start}' and time < '{self.end}' and site_id='{self.site}'"
+        interval = CONFIG["influxdb_client"]["interval"]
+        statement = f"SELECT {query_fields}" \
+                    f"FROM \"data-validation\" " \
+                    f"WHERE ({where}) GROUP BY time({interval}) ORDER BY desc"
+        return statement
+
+    @staticmethod
+    def extract_data(query):
+        """Extracts dataframe from the Influx database.
+
+        :param query: sql query to execute.
+        :return: Extracted data from the influx database.
+        :rtype: DataFrame.
+        """
+        client = DataFrameClient(host=CONFIG["influxdb_client"]["host"],
+                                 port=CONFIG["influxdb_client"]["port"],
+                                 username=os.environ['INFLUXDB_USERNAME'],
+                                 password=os.environ['INFLUXDB_PASSWORD'],
+                                 database=CONFIG["influxdb_client"]["database"],
+                                 ssl=CONFIG["influxdb_client"]["ssl"],
+                                 timeout=CONFIG["influxdb_client"]["timeout"])
```

Comment:
If these are the only keys in that dictionary, you could do this :)
```suggestion
        client = DataFrameClient(username=os.environ['INFLUXDB_USERNAME'],
                                 password=os.environ['INFLUXDB_PASSWORD'],
                                 **CONFIG["influxdb_client"]])
```

Reply:
thanks, going to try it. Im not sure if is going to work though seems `influxdb_client` have more configurations that the ones that `DataFrameClient` need, so not sure if its going to though an error, but I'm totally going to try it 👌

Reply:
actually there are more keys than these ones. The `client.query` option uses a couple of more keys from that config dict

Reply:
Yeah I actually see that now in the lines below this comment. Could be an argument for reorganizing the configuration, but this is totally fine for now.